source/Calamari/Kubernetes/SetupKubectlAuthentication.cs

```diff
@@ -46,148 +46,76 @@ public class SetupKubectlAuthentication
             this.workingDirectory = workingDirectory;
         }
 
-        public CommandResult Execute(string accountType)
+        public CommandResult Execute()
         {
-            var errorResult = new CommandResult(string.Empty, 1);
-
             foreach (var proxyVariable in ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables())
             {
                 environmentVars[proxyVariable.Key] = proxyVariable.Value;
             }
 
-            var kubeConfig = CreateKubectlConfig();
-            if (!kubectl.TrySetKubectl())
+            try
             {
-                return errorResult;
-            }
+                kubectl.SetKubectl();
+                var @namespace = DetermineNamespace();
+                SetupContext(@namespace);
+                CreateNamespace(@namespace);
 
-            var @namespace = variables.Get(SpecialVariables.Namespace);
-            if (string.IsNullOrEmpty(@namespace))
-            {
-                log.Verbose("No namespace provided. Using default");
-                @namespace = "default";
+                var outputKubeConfig = variables.GetFlag(SpecialVariables.OutputKubeConfig);
+                if (outputKubeConfig)
+                {
+                    kubectl.ExecuteCommandAndAssertSuccess("config", "view");
+                }
             }
-
-            if (!TrySetupContext(kubeConfig, @namespace, accountType))
+            catch (KubectlException e)
             {
-                return errorResult;
+                log.Error(e.Message);
+                return new CommandResult(string.Empty, 1);
             }
 
-            if (!CreateNamespace(@namespace))
-            {
-                log.Verbose("Could not create namespace. Continuing on, as it may not be working directly with the target.");
-            }
+            return new CommandResult(string.Empty, 0);
+        }
 
-            var outputKubeConfig = variables.GetFlag(SpecialVariables.OutputKubeConfig);
-            if (outputKubeConfig)
-            {
-                kubectl.ExecuteCommandAndAssertSuccess("config", "view");
-            }
+        string DetermineNamespace()
```

Comment:
nit: 
```suggestion
        string GetNamespaceOrDefault()
```

Might be more descriptive

Reply:
Makes sense :)